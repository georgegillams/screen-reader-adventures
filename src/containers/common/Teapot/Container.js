import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';
import { Subsection } from 'gg-components/Subsection';
import { Paragraph } from 'gg-components/Paragraph';
import PageTitle from 'components/common/PageTitle';

import STYLES from './teapot.scss';

const getClassName = cssModules(STYLES);

const Teapot = props => {
  const { className } = props;
  const classNameFinal = [getClassName('not-found__container')];
  if (className) {
    classNameFinal.push(className);
  }

  return (
    <main id="main" className={classNameFinal.join(' ')}>
      <PageTitle className={getClassName('not-found__container')} name="Error 418 - I'm a teapot." pageTitle="418">
        <Subsection anchor={false}>
          <Paragraph>
            Coffee cannot be brewed inside a teapot context. The resulting entity body MAY be short and stout.
          </Paragraph>
        </Subsection>
      </PageTitle>
    </main>
  );
};

Teapot.propTypes = {
  className: PropTypes.string,
};

Teapot.defaultProps = {
  className: null,
};

export default Teapot;
