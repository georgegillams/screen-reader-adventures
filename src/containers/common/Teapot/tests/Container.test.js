import React from 'react';
import { render } from '@testing-library/react';

import Teapot from '../Container';

describe('<Teapot />', () => {
  it('should render correctly', () => {
    const { container } = render(<Teapot />);

    expect(container).toMatchSnapshot();
  });
});
