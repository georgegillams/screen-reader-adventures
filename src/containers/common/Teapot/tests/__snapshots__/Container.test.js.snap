// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Teapot /> should render correctly 1`] = `
<div>
  <main
    class="not-found__container"
    id="main"
  >
    <h1
      class="page-title__heading not-found__container"
    >
      Error 418 - I'm a teapot.
    </h1>
    <div
      class="subsection__outer"
    >
      <span
        class="paragraph"
      >
        Coffee cannot be brewed inside a teapot context. The resulting entity body MAY be short and stout.
      </span>
    </div>
  </main>
</div>
`;
